### DISCUSSION TITLE
How do I add a space or transparent whitespace after each segment with style diamond?

### DISCUSSION THREAD

**Discussion by Vib1240n** [JanDeDobbeleer/oh-my-posh#4913] [Troubleshoot]
So I just installed oh my posh and my shell is ZSH, I am trying to figure out how can I add spaces after each segment 

<img width="903" alt="Screenshot 2024-05-06 at 9 14 35 PM" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/71455013/dedfcb52-5e61-41bf-9d2f-eba45eb83b18">

As you can see, there is no space after username, path and execution time. I tried to figure out to add whitespace after execution time and status but it just comes out weird. 

I am still just playing around with the code and just playing around with the JanDeDobbeleer.omp.json theme and this is what the modified code looks like. 
```{
    "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "blocks": [
        {
            "alignment": "left",
            "segments": [
                {
                    "background": "#c386f1",
                    "foreground": "#000000",
                    "leading_diamond": "\ue0b6",
                    "style": "diamond",
                    "template": " {{ .UserName }}",
                    "trailing_diamond": "\ue0b0",
                    "type": "session"
                },
                {
                    "background": "#ff479c",
                    "foreground": "#000000",
                    "properties": {
                        "home_icon": "~",
                        "style": "folder"
                    },
                    "style": "diamond",

                    "powerline_symbol": "\ue0b0",
                    "trailing_diamond": "\ue0b0",
                    "template": " \uea83{{ .Path }}",
                    "type": "path"
                },
                {
                    "background": "#fffb38",
                    "background_templates": [
                        "{{ if or (.Working.Changed) (.Staging.Changed) }}#FF9248{{ end }}",
                        "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#ff4500{{ end }}",
                        "{{ if gt .Ahead 0 }}#B388FF{{ end }}",
                        "{{ if gt .Behind 0 }}#B388FF{{ end }}"
                    ],
                    "foreground": "#193549",
                    "leading_diamond": "\ue0b6",
                    "powerline_symbol": "\ue0b0",
                    "properties": {
                        "branch_max_length": 25,
                        "fetch_stash_count": true,
                        "fetch_status": true,
                        "fetch_upstream_icon": true
                    },
                    "style": "powerline",
                    "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \ueb4b {{ .StashCount }}{{ end }} ",
                    "trailing_diamond": "\ue0b4",
                    "type": "git"
                },
                {
                    "background": "#ffff66",
                    "foreground": "#111111",
                    "powerline_symbol": "\ue0b0",
                    "style": "powerline",
                    "template": " \uf0ad ",
                    "type": "root"
                },
                {
                    "background": "#B4A2D6",
                    "foreground": "#000000",
                    "properties": {
                        "always_enabled": true
                    },
                    "style": "diamond",
                    // "leading_diamond": "\ue0b1",
                    "powerline_symbol": "\ue0b0",
                    "trailing_diamond": "\ue0b0",
                    "template": " {{ .FormattedMs }} ",
                    "type": "executiontime"
                },
                {
                    "background": "#23E9A4",
                    "background_templates": [
                        "{{ if gt .Code 0 }}#e91e63{{ end }}"
                    ],
                    "foreground": "#000000",
                    "properties": {
                        "always_enabled": true
                    },
                    "style": "diamond",
                    "template": "<parentBackground></> \ue23a ",
                    "trailing_diamond": "\ue0b0",
                    "leading_diamond": "\ue0b0",
                    "type": "status"
                },
                {
                    "background": "#0077c2",
                    "foreground": "#000000",
                    "style": "diamond",
                    // "leading_diamond": "\ue0b5",
                    "trailing_diamond": "\ue0b0",
                    "powerline_symbol": "\ue0b0",
                    "invert_powerline": true,
                    "template": "<#0077c2,transparent></>{{ .Name }} <transparent,#0077c2></>",
                    "type": "shell"
                },
                {
                    "background": "#1BD760",
                    "foreground": "#111111",
                    "invert_powerline": true,
                    "powerline_symbol": "\ue0b2",
                    "properties": {
                        "paused_icon": "\uf04c ",
                        "playing_icon": "\uf04b "
                    },
                    "style": "powerline",
                    "template": " \uf167 {{ .Icon }}{{ if ne .Status \"stopped\" }}{{ .Artist }} - {{ .Track }}{{ end }} ",
                    "type": "ytm"
                },
                {
                    "background": "transparent",
                    "foreground": "#ffffff",
                    "invert_powerline": true,
                    "leading_diamond": "\ue0b2",
                    "powerline_symbol": "\ue0b2",
                    "style": "diamond",
                    "template": " {{ .CurrentDate | date .Format }} ",
                    "trailing_diamond": "\ue0b4",
                    "type": "time"
                }
            ],
            "type": "prompt"
        },
        {
            "alignment": "left",
            "newline": true,
            "segments": [
                {
                    "style": "plain",
                    "template": "\u2570\u2500",
                    "type": "text"
                }
            ],
            "type": "prompt"
        }
    ],
    "console_title_template": "{{ .Shell }} in {{ .Folder }}",
    "final_space": false,
    "version": 2
}
```
```

**Comment by JanDeDobbeleer** [comment]
```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "background": "#c386f1",
          "foreground": "#000000",
          "leading_diamond": "\ue0b6",
          "style": "diamond",
          "template": " {{ .UserName }} ",
          "trailing_diamond": "\ue0b0",
          "type": "session"
        },
        {
          "background": "#ff479c",
          "foreground": "#000000",
          "properties": {
            "home_icon": "~",
            "style": "folder"
          },
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " \uea83 {{ .Path }} ",
          "type": "path"
        },
        {
          "background": "#fffb38",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#FF9248{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#ff4500{{ end }}",
            "{{ if gt .Ahead 0 }}#B388FF{{ end }}",
            "{{ if gt .Behind 0 }}#B388FF{{ end }}"
          ],
          "foreground": "#193549",
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "properties": {
            "branch_max_length": 25,
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          },
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \ueb4b {{ .StashCount }}{{ end }} ",
          "type": "git"
        },
        {
          "background": "#ffff66",
          "foreground": "#111111",
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " \uf0ad ",
          "type": "root"
        },
        {
          "background": "#B4A2D6",
          "foreground": "#000000",
          "properties": {
            "always_enabled": true
          },
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " {{ .FormattedMs }} ",
          "type": "executiontime"
        },
        {
          "background": "#23E9A4",
          "background_templates": [
            "{{ if gt .Code 0 }}#e91e63{{ end }}"
          ],
          "foreground": "#000000",
          "properties": {
            "always_enabled": true
          },
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": "<parentBackground></> \ue23a ",
          "type": "status"
        },
        {
          "background": "#0077c2",
          "foreground": "#000000",
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " {{ .Name }} ",
          "type": "shell"
        },
        {
          "background": "#1BD760",
          "foreground": "#111111",
          "properties": {
            "paused_icon": "\uf04c ",
            "playing_icon": "\uf04b "
          },
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " \uf167 {{ .Icon }}{{ if ne .Status \"stopped\" }}{{ .Artist }} - {{ .Track }}{{ end }} ",
          "type": "ytm"
        },
        {
          "background": "transparent",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "style": "diamond",
          "leading_diamond": "<transparent,background>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " {{ .CurrentDate | date .Format }} ",
          "type": "time"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "style": "plain",
          "template": "\u2570\u2500",
          "type": "text"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "{{ .Shell }} in {{ .Folder }}",
  "final_space": false,
  "version": 2
}
```

**Comment by KamiALK** [comment]
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "background": "#0077c2",
          "foreground": "#ffffff",
          "leading_diamond": "\u256d\u2500\ue0b6",
          "trailing_diamond": "\ue0b0",
          "style": "diamond",
          "template": " \uf120 {{ .Name }} ",
          "type": "shell"
        },
        {
          "background": "#ef5350",
          "foreground": "#FFFB38",
          "powerline_symbol": "\ue0b0",
          "style": "powerline",
          "template": " \uf292 ",
          "type": "root"
        },
        {
          "background": "#FF9248",
          "foreground": "#2d3436",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "folder_icon": " \uf07b ",
            "home_icon": "\ue617",
            "style": "folder"
          },
          "style": "powerline",
          "template": " \uf07b\uea9c {{ .Path }} ",
          "type": "path"
        },
        {
          "background": "#FFFB38",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#ffeb95{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#c5e478{{ end }}",
            "{{ if gt .Ahead 0 }}#C792EA{{ end }}",
            "{{ if gt .Behind 0 }}#C792EA{{ end }}"
          ],
          "foreground": "#011627",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "branch_icon": "\ue725 ",
            "fetch_status": true,
            "fetch_upstream_icon": true
          },
          "style": "powerline",
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }}<#ef5350> \uf046 {{ .Staging.String }}</>{{ end }} ",
          "type": "git"
        },
        {
          "background": "#83769c",
          "foreground": "#ffffff",
          "leading_diamond": "\ue0d7",
          "properties": {
            "style": "roundrock",
            "threshold": 0
          },
          "trailing_diamond": "\ue0b4",
          "style": "diamond",
          "template": " \ueba2 {{ .FormattedMs }}\u2800 ",
          "type": "executiontime"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "right",
      "segments": [
        {
          "background": "#303030",
          "foreground": "#3C873A",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "properties": {
            "fetch_package_manager": true,
            "npm_icon": " <#cc3a3a>\ue5fa</> ",
            "yarn_icon": " <#348cba>\ue6a7</>"
          },
          "style": "diamond",
          "template": "\ue718 {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }}",
          "type": "node"
        },
        {
          "background": "#306998",
          "foreground": "#FFE873",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue235 {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }}",
          "type": "python"
        },
        {
          "background": "#0e8ac8",
          "foreground": "#ffffff",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue738 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "java"
        },
        {
          "background": "#0e0e0e",
          "foreground": "#0d6da8",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue77f {{ if .Unsupported }}\uf071{{ else }}{{ .Full }}{{ end }}",
          "type": "dotnet"
        },
        {
          "background": "#ffffff",
          "foreground": "#06aad5",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue626 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "go"
        },
        {
          "background": "#f3f0ec",
          "foreground": "#925837",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue7a8 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "rust"
        },
        {
          "background": "#e1e8e9",
          "foreground": "#055b9c",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue798 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "dart"
        },
        {
          "background": "#ffffff",
          "foreground": "#ce092f",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue753 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "angular"
        },
        {
          "background": "#ffffff",
          "foreground": "#de1f84",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\u03b1 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "aurelia"
        },
        {
          "background": "#1e293b",
          "foreground": "#ffffff",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "{{ if .Error }}{{ .Error }}{{ else }}Nx {{ .Full }}{{ end }}",
          "type": "nx"
        },
        {
          "background": "#945bb3",
          "foreground": "#359a25",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "<#ca3c34>\ue624</> {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "julia"
        },
        {
          "background": "#ffffff",
          "foreground": "#9c1006",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue791 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "ruby"
        },
        {
          "background": "#ffffff",
          "foreground": "#5398c2",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\uf104<#f5bf45>\uf0e7</>\uf105 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}",
          "type": "azfunc"
        },
        {
          "background": "#565656",
          "foreground": "#faa029",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4 ",
          "style": "diamond",
          "template": "\ue7ad {{.Profile}}{{if .Region}}@{{.Region}}{{end}}",
          "type": "aws"
        },
        {
          "background": "#316ce4",
          "foreground": "#ffffff",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b4",
          "style": "diamond",
          "template": "\uf308 {{.Context}}{{if .Namespace}} :: {{.Namespace}}{{end}}",
          "type": "kubectl"
        },
        {
          "background": "#b2bec3",
          "foreground": "#222222",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "<transparent,background>\ue0b2</>",
          "properties": {
            "linux": "\ue712",
            "macos": "\ue711",
            "windows": "\ue70f"
          },
          "style": "diamond",
          "template": " {{ if .WSL }}WSL at {{ end }}{{.Icon}} ",
          "type": "os"
        },
        {
          "background": "#f36943",
          "background_templates": [
            "{{if eq \"Charging\" .State.String}}#b8e994{{end}}",
            "{{if eq \"Discharging\" .State.String}}#fff34e{{end}}",
            "{{if eq \"Full\" .State.String}}#33DD2D{{end}}"
          ],
          "foreground": "#262626",
          "invert_powerline": true,
          "powerline_symbol": "\ue0b2",
          "properties": {
            "charged_icon": "\uf240 ",
            "charging_icon": "\uf1e6 ",
            "discharging_icon": "\ue234 "
          },
          "style": "powerline",
          "template": " {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}\uf295 ",
          "type": "battery"
        },
        {
          "background": "#40c4ff",
          "foreground": "#ffffff",
          "invert_powerline": true,
          "leading_diamond": "\ue0b2",
          "trailing_diamond": "\ue0b4",
          "properties": {
            "time_format": "_2,15:04"
          },
          "style": "diamond",
          "template": " \uf073 {{ .CurrentDate | date .Format }} ",
          "type": "time"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "foreground": "#21c7c7",
          "style": "plain",
          "template": "\u2570\u2500",
          "type": "text"
        },
        {
          "foreground": "#e0f8ff",
          "foreground_templates": [
            "{{ if gt .Code 0 }}#ef5350{{ end }}"
          ],
          "properties": {
            "always_enabled": true
          },
          "style": "plain",
          "template": "\ue285\ueab6 ",
          "type": "status"
        }
      ],
      "type": "prompt"
    }
  ],
  "version": 3
}
